### PR TITLE
Added folders torch/csrc/onnx and torch/csrc/cuda to clang-tidy. Fixed some source issues in csrc/cuda.

### PR DESCRIPTION
**Pull request by shahofblah**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #74764

### CODE REVIEW

File: tools/linter/clang_tidy/__main__.py [hunk 1]
```diff
@@ -93,7 +94,13 @@ def clang_search_dirs() -> List[str]:
         "-torch/csrc/deploy/test_deploy_python_ext.cpp",
     ],
     "paths": ["torch/csrc/"],
-    "include-dir": ["/usr/lib/llvm-11/include/openmp"] + clang_search_dirs(),
+    "include-dir": [
+        "/usr/lib/llvm-11/include/openmp",
+        # We want the string '/usr/local/include/python<version number>'
+        # https://stackoverflow.com/a/35071359
+        gp()['include'],
+        "/__w/pytorch/pytorch/third_party/pybind11/include"
```
Comment (atalman) [inline tools/linter/clang_tidy/__main__.py:102]: Here please replace /__w/pytorch/pytorch/third_party/pybind11/include by  : os.path.join(PYTORCH_ROOT, "third_party/pybind11/include"),

File: tools/linter/clang_tidy/__main__.py [hunk 2]
```diff
@@ -93,7 +94,13 @@ def clang_search_dirs() -> List[str]:
         "-torch/csrc/deploy/test_deploy_python_ext.cpp",
     ],
     "paths": ["torch/csrc/"],
-    "include-dir": ["/usr/lib/llvm-11/include/openmp"] + clang_search_dirs(),
+    "include-dir": [
+        "/usr/lib/llvm-11/include/openmp",
+        # We want the string '/usr/local/include/python<version number>'
+        # https://stackoverflow.com/a/35071359
```
Comment (atalman) [inline tools/linter/clang_tidy/__main__.py:100]: please put it in a function call it something like:
`def get_python_include_dir() -> str: return gp()['include'] `

File: tools/linter/clang_tidy/run.py [hunk 3]
```diff
@@ -264,12 +264,16 @@ async def _run_clang_tidy(
     # Apply per-file options
     commands = []
     for f in files:
+        print(">>><<< processing file :")
```
Comment (atalman) [inline tools/linter/clang_tidy/run.py]: Please remove debuging print code

**Review by atalman — APPROVED** [review]

File: tools/linter/clang_tidy/__main__.py (hunk 1 above)
Comment (shahofblah) [reply tools/linter/clang_tidy/__main__.py]: Done

File: tools/linter/clang_tidy/__main__.py (hunk 2 above)
Comment (shahofblah) [reply tools/linter/clang_tidy/__main__.py]: Done

File: tools/linter/clang_tidy/run.py (hunk 3 above)
Comment (shahofblah) [reply tools/linter/clang_tidy/run.py]: Done

**Pull request merged**